doc/architecture.rst

```diff
@@ -369,3 +369,37 @@ is renamed or moved.
 Example:
 
   <oc:id>00000020oc5cfy6qqizm</oc:id>
+  
+Virtual Files
+----------------
+.. note::
+    * This feature is currently only available on ``Windows`` by default. ``Linux`` implementation is experimental and must be enabled by adding ``enableExperimentalOptions=true`` to the ``nextcloud.cfg`` configuration file in the ``App Data`` folder. ``macOS`` implementation is not available at the moment.
+
+Oftentimes, users are working with a huge amount of files that are big in size. Synchronizing every such file to a device that's running a Nextcloud desktop client is not always possible due to the user's device storage space limitation.
+Let's assume that your desktop client is connected to a server that has 1TB of data. Now, let's assume, you want all those files at hand, so you can quickly access any file via the file explorer. Your device has 512GB SSD storage inside.
```

Comment:
I would remove the second 'let's assume':
"Let's assume that your desktop client is connected to a server that has 1TB of data. You want all those files at hand, so you can quickly access any file via the file explorer. Your device has 512GB SSD storage inside."

Reply:
> Thanks
> I am not sure if we should not use another work to describe the hydration process. To my non native speaker, that looks terribly like a technical word.

@mgallien I like the word because it's short. But, I've now added an explanation within the parenthesis at the first occurrence of this, rather complex term :)